src/server/editorServices.ts

```diff
@@ -1372,24 +1372,50 @@ namespace ts.server {
             this.projectToSizeMap.forEach(val => (availableSpace -= (val || 0)));
 
             let totalNonTsFileSize = 0;
+
             for (const f of fileNames) {
                 const fileName = propertyReader.getFileName(f);
                 if (hasTypeScriptFileExtension(fileName)) {
                     continue;
                 }
+
                 totalNonTsFileSize += this.host.getFileSize(fileName);
+
                 if (totalNonTsFileSize > maxProgramSizeForNonTsFiles) {
+                    logExceedLimit.call(this);

```

Comment:
do we need top 5? or can we jsut log the largest so far? if so, we can keep the largest file in a local.

Reply:
W can do that too. I initially do top 5 just to get more information so I can exclude/fix multiple files at the same time.